root/usr/lib64/ruby/vendor_ruby/discovery/menu.rb

```diff
@@ -82,12 +82,12 @@ def configure_network static, mac, ip=nil, gw=nil, dns=nil, vlan=nil
   else
     command("nm-configure primary '#{mac}' '#{vlan}'")
   end
-  # NetworkManager monitors config files and auto connects immediately
   wait = cmdline('fdi.nmwait', 120)
-  result = command("nm-online -s -q --timeout=#{wait}")
-  # restarting proxy with regenerated SSL self-signed cert
+  command("nmcli -w #{wait} connection reload", false)
+  command("nmcli -w #{wait} connection up primary", false)
+  command("nm-online -s -q --timeout=#{wait}")
+  # wait for IPv4, generate SSL self-signed cert and start proxy
   command("systemctl start foreman-proxy")
-  result
```

Comment:
we used to return result of a nm-online command which makes sense to me for a method called `configure_network`, shouldn't we store the result on line 88 and keep the result here?

Reply:
Let me rewrite it and take both proxy and up command results into effect. Will do build for you shortly.